src/main/java/filter/expression/Qualifier.java

```diff
@@ -93,6 +92,47 @@ public Qualifier(String name, List<SortKey> keys) {
         this.sortKeys = new ArrayList<>(keys);
     }
 
+    public static FilterExpression replaceMilestoneAliases(IModel model, FilterExpression expr) {
+        List<Qualifier> metaQualifiers = expr.find(Qualifier::isMetaQualifier);
+        List<String> repoIds = getReposInMetaQualifiers(metaQualifiers);
+
+        if (repoIds.isEmpty()) {
+            repoIds.add(model.getDefaultRepo().toLowerCase());
+        }
+
+        List<TurboMilestone> allMilestones = model.getMilestones().stream()
+                .filter(ms -> repoIds.contains(ms.getRepoId().toLowerCase()))
+                .collect(Collectors.toList());
+        Collections.sort(allMilestones, Qualifier::milestoneDueDateComparator);
+
+        int currentMilestoneIndex = getCurrentMilestoneIndex(allMilestones);
+
+        FilterExpression exprWithoutMilestones = expr.filter(q -> !Qualifier.isMilestoneQualifier(q));
+        List<Qualifier> milestoneQualifiers = expr.find(Qualifier::isMilestoneQualifier);
+        List<Qualifier> convertedMilestoneQualifiers = new ArrayList<Qualifier>();
+
+        milestoneQualifiers.stream()
+                .filter(msQualifier -> msQualifier.getContent().isPresent())
+                .forEach(msQualifier -> {
+                    convertedMilestoneQualifiers.add(
+                            msQualifier.replaceMilestoneAliasQualifier(allMilestones, currentMilestoneIndex)
+                    );
+                });
+
+        for (Qualifier msQualifier : convertedMilestoneQualifiers) {
+            exprWithoutMilestones = new Conjunction(msQualifier, exprWithoutMilestones);
```

Comment:
This can change semantics: `(milestone:curr+1 && is:pr) || (milestone:curr && is:issue)` will become `(is:pr || is:issue) && milestone:a && milestone:b`, which isn't equivalent.

I think a better approach would be implement `map` for FilterExpressions.